exonum/src/blockchain/mod.rs

```diff
@@ -141,11 +142,14 @@ impl Blockchain {
     ///
     /// - Blockchain has service with the `service_id` of given raw message.
     /// - Service can deserialize given raw message.
-    pub fn tx_from_raw(&self, raw: RawMessage) -> Option<Box<Transaction>> {
+    pub fn tx_from_raw(&self, raw: RawMessage) -> Result<Box<Transaction>, MessageError> {
         let id = raw.service_id() as usize;
-        self.service_map.get(id).and_then(|service| {
-            service.tx_from_raw(raw).ok()
-        })
+        self.service_map
+            .get(id)
+            .map(|service| service.tx_from_raw(raw))
+            .unwrap_or_else(|| {
+                Err(MessageError::IncorrectServiceId { service_id: id as u16 })
```

Comment:
Technically speaking, this is incorrect use of `IncorrectServiceId`; its field denotes *expected* `service_id` and used as such in parsers inferred by the `messages!` / `transactions!` macro.